src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -239,24 +239,23 @@ public void updateFilteredVolunteerList(Predicate<Volunteer> predicate) {
         filteredVolunteers.setPredicate(predicate);
     }
 
-    @Override
-    public boolean equals(Object other) {
-        if (other == this) {
-            return true;
-        }
-
-        // instanceof handles nulls
-        if (!(other instanceof ModelManager)) {
-            return false;
-        }
+    /**
+     * Filters the list of volunteers to only include those whose names contain the specified search string,
+     * ignoring case. Updates the filtered volunteer list in the model to reflect this criteria.
+     *
+     * @param searchString The string to search for within volunteer names. This search is case-insensitive.
+     * @return An observable list of volunteers that contain the search string in their names.
+     */
+    public ObservableList<Volunteer> filterVolunteersByName(String searchString) {
+        Predicate<Volunteer> volunteerContainsSearchString = volunteer ->
+                volunteer.getName().toString().toLowerCase().contains(searchString.toLowerCase());
+        updateFilteredVolunteerList(volunteerContainsSearchString);
 
-        ModelManager otherModelManager = (ModelManager) other;
-        return addressBook.equals(otherModelManager.addressBook)
-                && userPrefs.equals(otherModelManager.userPrefs)
-                && filteredVolunteers.equals(otherModelManager.filteredVolunteers)
-                && filteredEvents.equals(otherModelManager.filteredEvents);
+        return getFilteredVolunteerList();
```

Comment:
This might be better as return void, you can maybe create a predicate here and use the setPredicate function within model to filter out the list instead. As mentioned earlier its probably better to not take the event/volunteer list out of model unless u really have to.